### DISCUSSION TITLE
Make SmartDate a class

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3477] [Ideas]
If memory serves we wanted to keep SmartDate a struct for compatibility with previous versions of Csla and also to keep it similar to .NET DateTime.
I am of the opinion we should change SmartDate to a class, it will make it simple to use the Text property of SmartDate from UI frameworks.

Regards.

**Comment by rockfordlhotka** [comment]
The design intent is that a business class would expose a string property that is used to interact with the private `SmartDate` instance's `Text` property, not to bind the `SmartDate` directly to the UI control.
